com.zsmartsystems.zigbee.dongle.zstack.autocode/src/main/java/com/zsmartsystems/zigbee/dongle/zstack/autocode/CommandGenerator.java

```diff
@@ -0,0 +1,1095 @@
+                    out.println("                builder.append(' ');");
+                    out.println("            }");
+
+                    String format = "%02X";
+                    if (parameter.displayType != null) {
+                        switch (parameter.displayType.toLowerCase()) {
+                            case "hex":
+                                String size = "";
+                                if (parameter.displayLength != 0) {
+                                    size = "0" + parameter.displayLength;
+                                }
+                                format = "%" + size + "X";
+                                break;
+                        }
+                    }
+
+                    out.println("            builder.append(String.format(\"" + format + "\", "
+                            + camelCaseToLowerCamelCase(parameter.name) + "[c]));");
+
+                    //
+                    // out.println(" builder.append(String.format(\"%02X\", " + formatParameterString(parameter)
+                    // + "[c]));");
+                    out.println("        }");
+                } else {
+                    out.println("        builder.append(" + formatParameterString(parameter) + ");");
+                }
+            }
+            out.println("        builder.append(']');");
+            out.println("        return builder.toString();");
+        }
+        out.println("    }");
+
+        out.println("}");
+
+        out.flush();
+
+        String cmdPackage = zstackCommandPackage + "." + command.subsystem.toLowerCase().replace("_", "");
+        File packageFile = new File(sourceRootPath + cmdPackage.replace(".", "/"));
+        packageFile.mkdirs();
+        PrintWriter outFile = getClassOut(packageFile, className);
+
+        outputCopywrite(outFile);
+        outFile.println("package " + cmdPackage + ";");
+
+        outFile.println();
+
+        outputImports(outFile);
+
+        outFile.println();
+        outFile.print(stringWriter.toString());
+
+        outFile.flush();
+        outFile.close();
+
+        out.close();
+    }
+
+    private void createStructureClass(Structure structure) throws FileNotFoundException {
+        String className = upperCaseFirstCharacter(structure.name);
+
+        System.out.println("Processing structure class " + structure.name + "  [" + className + "()]");
+
+        StringWriter stringWriter = new StringWriter();
+        PrintWriter out = new PrintWriter(stringWriter);
+
+        clearImports();
+
+        out.println("/**");
+        out.println(" * Class to implement the Z-Stack structure <b>" + structure.name + "</b>.");
+        out.println(" * <p>");
+        if (structure.description != null && !structure.description.isEmpty()) {
+            outputWithLinebreak(out, "", structure.description);
+            out.println(" * <p>");
+        }
+        out.println(" * Note that this code is autogenerated. Manual changes may be overwritten.");
```

Comment:
I'd suggest to remove the @author tag below and instead add the `@Generated` annotation.

```
@Generated(value = "com.zsmartsystems.zigbee.autocode.ZigBeeCodeGenerator", date = "2020-12-25T10:11:19Z")
```